lib/routes/account.js

```diff
@@ -1733,6 +1755,15 @@ module.exports = function (
         var ip = request.app.clientAddress
         var emailRecord
 
+        // Store flowId and flowBeginTime to send in email
+        let flowId, flowBeginTime
+        if (request.payload.metricsContext) {
+          flowId = request.payload.metricsContext.flowId
+          flowBeginTime = request.payload.metricsContext.flowBeginTime
+        }
+
+        request.validateMetricsContext()
```

Comment:
We should move this call to `validateMetricsContext` before we store `flowId` and `flowBeginTime`, otherwise we'll emit events containing bad data.